### DISCUSSION TITLE
Add clean_points() for Gradient

### DISCUSSION THREAD

**Discussion by CsloudX** [godotengine/godot-proposals#7192] [Engine Core]
### Describe the project you are working on

/

### Describe the problem or limitation you are having in your project

I want use `Gradient` Class in GDScript, and I found it has 2 default points in `Gradient`. I want use my color and try clean the default points.
I'm try 3 ways to do that:
1.
```
gradient.offsets.clear()
gradient.colors.clear()
```
2.
```
gradient.remove_point(0)
gradient.remove_point(1)
```
3.
```
while gradient.get_point_count()>0:
	gradient.remove_point(0)
```
But all of them can't work.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

/

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

/

### If this enhancement will not be used often, can it be worked around with a few lines of script?

/

### Is there a reason why this should be core and not an add-on in the asset library?

/

**Comment by AThousandShips** [comment]
You have to fill in all the questions in the form, they are not optional, if you cannot open a [discussion](https://github.com/godotengine/godot-proposals/discussions) instead

**Comment by AThousandShips** [comment]
Also this makes no sense, gradients have to have at least one point, how else can it work?

Gradient prevents removing the last point by design, so this would go against that principle

**Reply by CsloudX** [reply]
I'm saved my colors to file, and want to restore my colors to `Gradient`. So I need clean the colors first.

**Reply by AThousandShips** [reply]
That seems like a very specific issue where you'd be better off not doing the save step if you're just gonna reset them anyway, just replace the gradient to a new one?

You can just use:
```gdscript
gradient.offsets = PackedFloat32Array()
```

The reason 1 fails in your example is that it doesn't return the data it copies it

**Reply by AThousandShips** [reply]
This does however raise a documentation issue, will update the documentation to clarify that the `offset` and `colors` cannot be directly modified but have to be assigned

Edit: Done https://github.com/godotengine/godot/pull/78970
